### ISSUE TITLE
Provide message ID in (debug|trace|verbose) output

### ISSUE THREAD

**Issue by abalage** [syslog-ng/syslog-ng#2828]
I tried to debug a issue about geoip parsing. Therefore I turned on debug, trace, verbose modes.
Checked the logs and mistakenly realized there is a msg='' field in almost all of the messages. I thought it is an ID for a given message.

I got realized that thing actually represents more messages not just one. After checking the source code I realized that ID is actually just a C pointer and not an actual ID. (printf + %p)

Do you have plans to help troubleshooting by adding a unique ID to any message throughout their lifetime?

**Comment by bazsi** [comment]
syslog-ng clones LogMessage instances as it traverses the processing pipeline, so that pointer is as close to the message id as we can get.

We do have a $RCPTID value for each message, but that denotes the ID of the message as it was received. That would stay the same across clones, but I am not sure this is what you want.

Hmm, now as I was reading you comment, maybe that problem is that this ID is reused over time?

**Comment by abalage** [comment]
Exactly. That ID could be useful but it getting reused for other messages too, effectively making troubleshooting hard.
Also I am sure there is a internal reason why cloning happens in the background, but it is not trivial to understand it from operations point of view.

**Comment by Kokan** [comment]
@abalage 
Would the `$RPCTID` be useful for you as an ID ?
As the above macro does not always exists (user must configure it), and we do not want to generate by default (because of performance).

Such a logic could exists that checks if `$RPCTID` is generated, if yes than uses that value as message ID, otherwise falls back to the current message pointer address

Would  a logic like that okay for you ?

**Comment by abalage** [comment]
@Kokan Did you mean this?

```

RCPTID

Description: When the use-rcptid global option is set to yes, syslog-ng OSE automatically assigns a unique reception ID to every received message. You can access this ID and use it in templates via the ${RCPTID} macro. The reception ID is a monotonously increasing 48-bit integer number, that can never be zero (if the counter overflows, it restarts with 1).
```
Is this macro gets logged without changing anything but only the global option?

**Comment by Kokan** [comment]
Yes, I meant that.

No that macro is not logged currently. There was a discussion to make a patch that works as I described above if that is okay for you.

**Comment by abalage** [comment]
If it isn't a big deal, then please do that. It is better than nothing.

**Comment by Kokan** [comment]
Do you have alternative suggestions ?
The bigger part of the work is finding and replacing all of the current msg pointer print usage (and that is independent of the ID definition).

**Comment by abalage** [comment]
Frankly I do not have. What I can add is to better explain the use case.

Me as an end user would like to troubleshoot complex configuration related issues. And I mean the configuration is complex, not the issue. So lots of rewrites, filters and so on.
Be able to track what path any given message passes throughout its lifetime. A unique ID would help a lot in tracking.

**Comment by Kokan** [comment]
Just for the record, a banch that introduces msg ids tag: https://github.com/Kokan/syslog-ng/tree/evt_log_msg_id_tag As it was a long time created the instance of msg pointer usage should be re-checked again latest. A PR from this only using the current pointery thing should be okay, as at least that introduces an evt tag for msg id, which later can be changed.

**Comment by MrAnno** [comment]
Implemented in #3972
